src/ConnectedMode/Migration/MSBuildFileCleaner.ElementAndAttributeTailMatcher.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Migration
+{
+    internal partial class MSBuildFileCleaner
```

Comment:
@bigfluffycookie I wanted to make the `ElementAndAttributeTailMatcher` helper class a `private` class, since it's only ever going to be used by the cleaner class (we're not testing it independently, just via the cleaner class).

To make it `private` it has to be nested inside another class, in this case the cleaner class.

However, I also want to put the code in a separate file. So that meant making the cleaner class `partial` so I could split its definition across multiple files and put the new helper class as a private  class inside one of them.